core/ecschema-metadata/src/test/Metadata/SchemaFromJson.test.ts

```diff
@@ -0,0 +1,334 @@
+            schemaItemType: "RelationshipClass",
+            strength: "Embedding",
+            strengthDirection: "Forward",
+            modifier: "Sealed",
+            customAttributes: [
+              {
+                className: "DummyReference.customAttributeOne",
+                showClasses: true,
+              },
+            ],
+            source: {
+              polymorphic: true,
+              multiplicity: "(0..*)",
+              roleLabel: "Source RoleLabel",
+              constraintClasses: [
+                "SchemaA.testClassOne",
+              ],
+            },
+            target: {
+              polymorphic: true,
+              multiplicity: "(0..*)",
+              roleLabel: "Target RoleLabel",
+              constraintClasses: [
+                "SchemaA.testClassOne",
+              ],
+            },
+          },
+          testClassOne: {
+            schemaItemType: "EntityClass",
+            description: "Test class one",
+          },
+          testClassTwo: {
+            schemaItemType: "EntityClass",
+            description: "Test class two",
+          },
+        },
+      };
+
+      await expect(Schema.fromJson(schemaA, context)).to.be.rejectedWith(ECObjectsError, "Unable to load custom attribute DummyReference.customAttributeOne from container SchemaA.relationshipOne, DummyReference reference not defined");
+    });
+  });
+
+  describe("Schema from json creation with property containers that have custom attributes tests", () => {
+    it("should throw an error and not allow the creation of the schema with custom attribute in a primitive property container", async () => {
+      const _dummyRefOne = await Schema.fromJson({
+        ...dummyRefJson,
+        items: {
+          customAttributeOne: {
+            schemaItemType: "CustomAttributeClass",
+            appliesTo: "AnyClass",
+          },
+        },
+      }, context);
+
+      const schemaA = {
+        ...schemaAJson,
+        items: {
+          testClassOne: {
+            schemaItemType: "EntityClass",
+            description: "Test class one",
+            properties: [
+              {
+                name: "Offset",
+                type: "PrimitiveProperty",
+                isReadOnly: true,
+                priority: 0,
+                typeName: "double",
+                customAttributes: [
+                  {
+                    className: "DummyReference.customAttributeOne",
+                    showClasses: true,
+                  },
+                ],
+              },
+            ],
+          },
+        },
+      };
+
+      await expect(Schema.fromJson(schemaA, context)).to.be.rejectedWith(ECObjectsError, "Unable to load custom attribute DummyReference.customAttributeOne from container testClassOne.Offset, DummyReference reference not defined");
+    });
+
+    it("should throw an error and not allow the creation of the schema with custom attribute in an array property container", async () => {
```

Comment:
I don't think you need this test.  There is no meaningful difference between types of property definitions when it comes to custom attribute behavior.

Reply:
Removed.